### PR TITLE
Pass file uris to compile

### PR DESCRIPTION
**Pull request by jiahuijiang**
It might be useful for some language servers

### CODE REVIEW

File: language-server-commons/src/main/java/com/palantir/ls/DefaultCompilerWrapper.java [hunk 1]
```diff
@@ -87,8 +87,8 @@ public URI getWorkspaceRoot() {
     }
 
     @Override
-    public Set<PublishDiagnosticsParams> compile() {
-        Set<PublishDiagnosticsParams> diagnostics = compiler.compile();
+    public Set<PublishDiagnosticsParams> compile(Set<URI> file) {
```
Comment (natgabb) [inline language-server-commons/src/main/java/com/palantir/ls/DefaultCompilerWrapper.java]: file -> files

File: language-server-commons/src/main/java/com/palantir/ls/api/WorkspaceCompiler.java [hunk 2]
```diff
@@ -38,7 +38,7 @@
      * Compiles all relevant files in the workspace.
      * @return the compilation warnings and errors by file
      */
-    Set<PublishDiagnosticsParams>  compile();
+    Set<PublishDiagnosticsParams> compile(Set<URI> file);
```
Comment (natgabb) [inline language-server-commons/src/main/java/com/palantir/ls/api/WorkspaceCompiler.java]: file -> files

Maybe update docs

**Comment by jiahuijiang** [comment]
haha thank you Natacha :)

**Comment by jiahuijiang** [comment]
@darora for your SA in case you want to only compile part of the repo

**Comment by darora** [comment]
Sweet, thanks, will be pretty nice

**Pull request merged**